File: 1dec/partone.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.*;

public class partone {
    public static void main(String[] args) {
        BufferedReader objReader = null;
        ArrayList<String> data = new ArrayList<String>();

        try {
        String strCurrentLine;
        objReader = new BufferedReader(new FileReader("C:/Users/samue/Documents/GitHub/aoc2021-smule/1dec/input.txt"));
        while ((strCurrentLine = objReader.readLine()) != null) {
            data.add(strCurrentLine);
        }
        } catch (IOException e) {
            e.printStackTrace();
        } finally {

        try {
            if (objReader != null)
            objReader.close();
        } catch (IOException ex) {
            ex.printStackTrace();
        }
        }

        int count = 0;
        for (int i = 0; i < data.size()-1; i++) {
            if (Integer.parseInt(data.get(i)) < Integer.parseInt(data.get(i+1))) {
                count += 1;
            }
        }
        System.out.println("count: " + count);
    }
}

File: 10dec/partone.py
with open("input.txt") as file:
    data = file.read().splitlines()
    count = 0
    for line in data:
        print(line)
        list = []
        for i in line:
            if i in ['(', '[', '{', '<']:
                list.append(i)
            elif i == ')':
                if list[-1] == '(':
                    list.pop()
                else:
                    count += 3
                    broke = True
                    break

            elif i == ']':
                if list[-1] == '[':
                    list.pop()
                else:
                    count += 57
                    broke = True
                    break

            elif i == '}':
                if list[-1] == '{':
                    list.pop()
                else:
                    count += 1197
                    broke = True
                    break

            elif i == '>':
                if list[-1] == '<':
                    list.pop()
                else:
                    count += 25137
                    broke = True
                    break
        
print(count)